README.md

```diff
@@ -1,64 +1,199 @@
-This project explains the necessary steps to combine CrateDB with Apache Spark, a lightning fast cluster system which offers a variety operations for BigData.
-A really poplar feature of Spark is machine learning, which will be covered here.
-This project explains the necessary steps to apply machine learning algorithms on big data.
-For storage and ingestion, CrateDB is the software to store and query an incredible amount of data in real time.
-For machine learning, Apache Spark is the quick and easy way to produce machine learning models.
+## About
 
+[CrateDB](http://crate.io) is a distributed SQL database system which is capable of managing huge amounts of data in real time.
+It is an easy, horizontal scalable, efficient and system independent solution built for IoT applications.
+
+Though CrateDB is a pretty solid choice for managing BigData, it was not built for performing general data analysis.
+This project adds this feature to CrateDB by creating a connection with Apache Spark as data processing tool.
+
+[Apache Spark](http://spark.apache.org) is a cluster computing engine which offers lots and lots of various data operations.
+All data operations are done in a distributed way, where all computations are done in RAM, which performs about 100x times faster than the previous Apache Hadoop system.
+A really poplar feature of Spark is machine learning, which will mainly be covered here.
 
 Since the theory may be complex at first sight, a simple use case helps to explain the theory behind.
 
-## Requirements
+### Known Issues and Limits
+
+For this project, it was not possible to solve every occuring problem at its source.
+But for almost every known problem a workarround could be made. 
+These problems may be fixed in future.
+
+#### No automatic creation of tables from Spark
+
+Since there is currently no CrateDB SQL Dialect implemented in JDBC it is not possible automatically create every table automatically.
+Therefore, every table in which data shall be ingested via Sparks JDBC interface must be created manually.
+
+The create table statements needed for this use case can be found in the section "Use Case Setup".
+
+#### No streaming between CrateDB and Spark
+
+At the moment writing this document it is not possible to create a streaming application due to following reasons:
+
+* Spark does not yet support streaming via JDBC
+* Sparks machine learning algorithms are currently not made for stream-based applications
+
+One possible way to get around this issue is to fetch only data which has no predictions already.
+This job needs to be executed at regular intervals then.
+
+#### Persisting a machine learning model in a cluster
+
+Since the computations done in a cluster are distributed, also the persistence must be in a distributed or a central way.
+The supported distributed ways to persist in Spark is via hdfs protocol.
+A central approach would be for example s3a protocol.
```

Comment:
I think this section should be a little bit clearer. It doesn't cover that you now found a solution using s3. Would also be interesting to reason why using CrateDB to store the model was not possible.